webapp/components/survey/SurveyForm/nodeDefs/components/types/NodeDefCode/store/hooks/useItems.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef, useState } from 'react'
+
+import * as API from '@webapp/service/api'
+
+export const useItems = ({ categoryUuid, categoryLevelIndex, draft, edit, parentCategoryItemUuid, surveyId }) => {
+  const [items, setItems] = useState([])
+  const cancelRequestRef = useRef(null)
+
+  // Fetch category items on categoryUuid or nodeParentCodeUuid update
+  useEffect(() => {
+    if (!edit) {
+      const { request: fetchItems, cancel } =
+        categoryUuid && (parentCategoryItemUuid || categoryLevelIndex === 0)
+          ? API.fetchCategoryItems({ surveyId, categoryUuid, draft, parentUuid: parentCategoryItemUuid })
+          : []
+      cancelRequestRef.current = cancel
+
+      fetchItems
+        .then(({ data: { items: itemsLoaded = {} } }) => {
+          setItems(itemsLoaded)
+        })
+        .catch(() => {
+          // canceled
+        })
```

Comment:
how about use async-await? also, the fetch on line 14 is async

Reply:
we cannot "await" for the promise to resolve, otherwise we won't be able to cancel it when needed.